src/Bridges/Constraint/indicator.jl

```diff
@@ -51,3 +51,89 @@ function concrete_bridge_type(::Type{<:IndicatorActiveOnFalseBridge},
                               ::Type{MOI.IndicatorSet{MOI.ACTIVATE_ON_ZERO, S}}) where {F<:MOI.VectorAffineFunction, S<:MOI.AbstractScalarSet}
     return IndicatorActiveOnFalseBridge{Float64, F, S}
 end
+
+## Indicator constraint by SOS1 bridge
+
+"""
+    IndicatorSOS1Bridge{T, BC <: Union{MOI.LessThan{T}, MOI.GreaterThan{T}, MOI.EqualTo{T}}}
+
+The `IndicatorSOS1Bridge` replaces an indicator constraint of the following form:
+``z \\in \\mathbb{B}, z == 1 \\implies f(x) \\leq b`` with a SOS1 constraint:
+``z \\in \\mathbb{B}, w \\leq 0, f(x) + w \\leq b, SOS1(w, z)``.
+`GreaterThan` constraints are handled in a symmetric way:
+``z \\in \\mathbb{B}, z == 1 \\implies f(x) \\geq b`` is reformulated as:
+``z \\in \\mathbb{B}, w \\geq 0, f(x) + w \\geq b, SOS1(w, z)``.
+`EqualTo` constraints are handled without a bound constraint:
+``z \\in \\mathbb{B}, z == 1 \\implies f(x) == b`` is reformulated as:
+``z \\in \\mathbb{B}, w \\text{ free}, f(x) + w == b, SOS1(w, z)``.
+
```

Comment:
Why not always add a free `w` so that it works for any indicator constraint (also `MOI.Interval`, ...) ?

Reply:
Can we have a `free` fallback with all constraints but `LessThan`, `GreaterThan`, otherwise we add a free, which has to be re-bounded by two variables for `LP`& `MILP` solvers

Reply:
Not sure I understood "which has to be re-bounded by two variables"

Reply:
Sorry yes, if we create the variable `w` free, it will be replaced by `w+ >= 0, w- >= 0, w = w+ + w- ` instead of creating a bounded `w` right away

Reply:
That's also the case for SDP solvers with the `Variable.FreeBridge`. Which LP solver does not support free variables ? This transformation is done internally ?
For a solver supporting free variables such constraints may increase the size of the problem, we should probably have *informative* constraints but that's out of the scope of this PR.

Reply:
> Which LP solver does not support free variables ? This transformation is done internally ?

All solvers (that I am aware of) support free variables, but the transformation needs to be done internally somehow. The simplex algorithm does require positive variables

Reply:
If you want to use the standard form you have equality constraints and nonnegative variables but then the dual has free variables. Some solvers might solve the dual using the simplex in which case they need all variables to be free. Although that's the textbook simplex so I don't know whether solvers might use a more sophisticated one supporting mixing free and non-free ones.

Reply:
> If you want to use the standard form you have equality constraints and nonnegative variables but then the dual has free variables.

My understanding is that the standard form is a form  `A x <= b, x >= 0` transformed in `Ax + s = b, x, s >= 0`. You need to have basic columns in the equality, so starting from a generic `A x = b`, it will need to change the variable system in `[A; -A] x <= [b; -b]` and then add slacks, in order to have as many basic columns as constraints.